src/screen/RequestScreen.js

```diff
@@ -236,14 +236,15 @@ class RequestScreen extends Screen {
 			referrer: 'client'
 		});
 
-		this.setRequest(request);
-
 		return Promise.race([
 			fetch(request)
 				.then(resp => {
 					this.removeSafariXHRPolyfill();
 					this.assertValidResponseStatusCode(resp.status);
 
+					request.responseURL = resp.url;
+					this.setRequest(request);
```

Comment:
Let's keep this line where it was just in case somewhere else is relying on it being set (because otherwise, if we end up `throw`-ing, we'll fail to set `request`).